### PR TITLE
[BugFix] Fix update statement add array literal cast expression error

### LINKED ISSUE #10009: unknown error when update primary table
### Steps to reproduce the behavior (Required)
1. create primary table:
```
create table pri_tbl(
col_tinyint tinyint
,col_date date
,col_char char(10)
,col_string string
,col_smallint smallint
,col_timestamp datetime
,col_varchar varchar(100)
,col_boolean boolean
,col_int int
,col_bigint bigint
,col_float float
,col_double double
,col_decimal decimal(38,18)
,col_largeint largeint
,col_array array<string>
,col_json json)
ENGINE=OLAP
primary KEY(col_tinyint,col_date)
DISTRIBUTED BY HASH(col_tinyint) BUCKETS 3
PROPERTIES ("replication_num" = "3");
```

2. update:
```
mysql> update pri_tbl set col_array = ['231','4321','42']  where col_date < '2022-10-10';
ERROR 1064 (HY000): Unknown error
```

fe.warn.log:
```
2022-08-15 20:01:32,067 WARN (starrocks-mysql-nio-pool-138|64520) [StmtExecutor.execute():490] execute Exception, sql update pri_tbl set col_array = ['231','4321','42']  where col_date < '2022-10-10'
java.lang.NullPointerException: null
        at com.starrocks.analysis.CastExpr.analyze(CastExpr.java:156) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.CastExpr.<init>(CastExpr.java:62) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.UpdateAnalyzer.analyze(UpdateAnalyzer.java:68) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitUpdateStatement(Analyzer.java:234) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitUpdateStatement(Analyzer.java:80) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.UpdateStmt.accept(UpdateStmt.java:55) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:126) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.analyze(Analyzer.java:82) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer.analyze(Analyzer.java:77) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:43) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:347) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:316) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:429) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:667) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_322]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_322]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_322]
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+--------------------+
| current_version()  |
+--------------------+
| 2.4.0-0815 f5c2bd7 |
+--------------------+
1 row in set (0.01 sec)
```

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10009 

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java [hunk 1]
```diff
@@ -82,5 +77,15 @@ public static void analyze(UpdateStmt updateStmt, ConnectContext session) {
 
         updateStmt.setTable(table);
         updateStmt.setQueryStatement(queryStatement);
+
+
+        List<Expr> outputExpression = queryStatement.getQueryRelation().getOutputExpression();
+        List<Expr> castOutputExpressions = Lists.newArrayList();
+        for (int i = 0; i < table.getBaseSchema().size(); ++i) {
```
Comment (decster) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java:84]: add a assertion or check `outputExpression.size() == table.getBaseSchema().size()` ?

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java:86]: done

**Review by decster — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|9|9|100.00%|[]|

**Pull request merged**